src/dotnet/src/Mil.Navy.Nrl.Norm/NormSession.cs

```diff
@@ -0,0 +1,918 @@
+            {
+                infoBytes = null;
+                infoLength = 0;
+            }
+            var objectHandle = NormFileEnqueue(_handle, filename, infoBytes, infoLength);
+            if (objectHandle == NormObject.NORM_OBJECT_INVALID)
+            {
+                throw new IOException("Failed to enqueue file");
+            }
+            return new NormFile(objectHandle);
+        }
+
+        /// <summary>
+        /// This function enqueues a segment of application memory space for transmission.
+        /// </summary>
+        /// <remarks>
+        /// This is an overload which will call DataEnqueue() with info set to null, infoOffset set to 0, and infoLength set to 0.
+        /// </remarks>
+        /// <param name="dataBuffer">The dataBuffer is a byte array containing the message to be transmitted.</param>
+        /// <param name="dataOffset">Indicates the start of the message. Anything before it will not be sent. 
+        /// Note: to send full message dataOffset should be set to 0.</param>
+        /// <param name="dataLength">Size of the message.</param>
+        /// <returns>A NormData is returned which the application may use in other NORM API calls as needed.</returns>
+        /// <exception cref="IOException">Thrown when NormDataEnqueue() returns NORM_OBJECT_INVALID, indicating the failure to enqueue data.</exception>
+        public NormData DataEnqueue(byte[] dataBuffer, int dataOffset, int dataLength)
+        {
+            return DataEnqueue(dataBuffer, dataOffset, dataLength, null, 0, 0);
+        }
+
+        /// <summary>
+        /// This function enqueues a segment of application memory space for transmission.
+        /// </summary>
+        /// <param name="dataBuffer">The dataBuffer is a byte array containing the message to be transmitted.</param>
+        /// <param name="dataOffset">Indicates the start of the message. Anything before it will not be sent. 
+        /// Note: to send full message dataOffset should be set to 0.</param>
+        /// <param name="dataLength">Size of the message.</param>
+        /// <param name="info">The optional info and infoLength parameters are used to associate NORM_INFO content with the sent transport object.</param>
+        /// <param name="infoOffset">Indicates the start of the message.</param>
+        /// <param name="infoLength">The optional info and infoLength parameters are used to associate NORM_INFO content with the sent transport object.</param>
+        /// <returns>A NormData is returned which the application may use in other NORM API calls as needed.</returns>
+        /// <exception cref="IOException">Thrown when NormDataEnqueue() returns NORM_OBJECT_INVALID, indicating the failure to enqueue data.</exception>
+        public NormData DataEnqueue(byte[] dataBuffer, int dataOffset, int dataLength, byte[]? info, int infoOffset, int infoLength)
+        {
+            var dataBytes = dataBuffer.Skip(dataOffset).Take(dataLength).ToArray();
```

Comment:
This is inefficient. But pointer arithmetic would work trivially.

Reply:
Refactored `DataEnqueue` to use pointer arithmetic instead of LINQ. Added tests for data offset, data length, info offset, and info length.